### ISSUE TITLE
Unable to connect to dashboard using external URL

### ISSUE THREAD

**Issue by SebastienBatteur** [home-assistant/iOS#3296]
Hello,

Since upgrading to version 2024.12 on my IPhone, I can no longer access the Home Assistant dashboard. I can authenticate with the external address but as soon as I'm logged in, it only displays the loading image. When I switch to my local network with the local url, the dashboard loads correctly. But as soon as I switch back to the mobile network, the dashboard disappears and returns to the loading image. 

I have no problem connecting via the browser to the external URL on the same phone.

I don't know if it's related but in the home assistant logs I found these when I connect with the external address
[36m2024-12-22 14:29:57.584   DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection]   [546540779504] Sebastien from 2a02:a020:5c3:36d2:cc76:6ae7:e169:65dc   (Mozilla/5.0 (iPhone; CPU iPhone OS 18_2 like Mac OS X) AppleWebKit/605.1.15   (KHTML, like Gecko) Home Assistant/2024.12.2 (io.robbie.HomeAssistant;   build:2024.1058; iOS 18.2.0) Mobile/HomeAssistant, like Safari): Sending   b'[{"id":27,"type":"result","success":false,"error":{"code":"config_not_found","message":"No   config found."}}]'[0m

when I log on locally, I find an id 27 included in the following message
[36m2024-12-22   14:29:45.883 DEBUG (MainThread)   [homeassistant.components.websocket_api.http.connection] [546489702656]   Sebastien from 10.X.X.X (Mozilla/5.0 (iPhone; CPU iPhone OS 18_2 like Mac   OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2024.12.2   (io.robbie.HomeAssistant; build:2024.1058; iOS 18.2.0) Mobile/HomeAssistant,   like Safari): Sending   b'[{"id":5,"type":"result","success":true,"result":{"entity_categories":{"0":"config","1":"diagnostic"},"entities":[{"ei":"update.home_assistant_supervisor_update","pl":"hassio","lb":  ......
.... {"id":27,"type":"result","success":true,"result":{"numeric_device_classes":["energy_storage","pressure","signal_strength","sound_pressure","ozone","volatile_organic_compounds","precipitation_intensity","moisture","frequency","data_size","volume_storage","pm25","voltage","nitrogen_dioxide","nitrous_oxide","carbon_monoxide","wind_speed","aqi","temperature","power","gas","humidity","irradiance","pm10","power_factor","battery","illuminance","atmospheric_pressure","ph","weight","energy","carbon_dioxide","blood_glucose_concentration","monetary","distance","apparent_power","volume","area","volume_flow_rate","reactive_power","volatile_organic_compounds_parts","conductivity","duration","nitrogen_monoxide","speed","sulphur_dioxide","precipitation","data_rate","current","water","pm1"]}}

if I look at the previous request with the external url: id 27 is not present (there are 26 and 28) which is the same as with the internal url

[36m2024-12-22 14:29:57.570   DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection]   [546540779504] Sebastien from 2a02:a020:5c3:36d2:cc76:6ae7:e169:65dc   (Mozilla/5.0 (iPhone; CPU iPhone OS 18_2 like Mac OS X) AppleWebKit/605.1.15   (KHTML, like Gecko) Home Assistant/2024.12.2 (io.robbie.HomeAssistant;   build:2024.1058; iOS 18.2.0) Mobile/HomeAssistant, like Safari): Sending   b'[{"id":3,"type":"result","success":true,"result":null},{"id":3,"type":"event","event":{"a":{"person.sebastien":{"s":"unknown","a":{"editable":true,"id":"sebastien","device_trackers":.....

I've already tried with another account on the phone. delete the local data on the phone, restore the device in home assistant, delete and reinstall the app but nothing changes.

in version 2024.11.x, access with the external URL worked correctly

Model Name: Iphone 13 Pro, IOS version 18.2
Software Version: 2024.12.2
App version: Core: 2024.12.5 - Supervisor: 2024.12.0 - user interface: 20241127.8

**Home Assistant Core Version**

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by bgoncal** [comment]
Whats your remote connection setup? Also export your companion app logs and submit it here: https://forms.gle/Uoqz127Phx4mMTpS6

**Comment by SebastienBatteur** [comment]
external access is via Cloudflare via the Cloudflared module, which refers to the ‘NGINX Home Assistant SSL proxy’ module because another app needs to be accessible with its own certificate.

the log file as soon as possible:

**Comment by SebastienBatteur** [comment]
companion app logs posted

**Comment by bgoncal** [comment]
According to the logs it was able to successfuly reach out to your remote URL and also login (as you mentioned before), so it could be some frontend cache involved, have you tried  opening companion App settings >> debuging >> reset frontend?

**Comment by SebastienBatteur** [comment]
Yes, before sending the extract logs, I cleaned up the event logs, reset the app entities database, reset the frontend cache before closing the app completely and opening it to extract the logs.

Initially that's what I thought too, that it was a cache problem. I even tried deleting the data in IOS and even deleting the app, nothing changed. 
but the strangest thing is that with the same account but with internet access everything works fine.
